qemu/tests/numa_stress.py

```diff
@@ -73,69 +51,70 @@ def run(test, params, env):
-        dd_cmd = "dd if=/dev/urandom of=%s bs=1k count=%s" % (memory_file,
-                                                              tmpfs_size)
-        numa_dd_cmd = "numactl -m %s %s" % (numa_node_malloc, dd_cmd)
-        error_context.context("Try to allocate memory in node %s"
-                              % numa_node_malloc, test.log.info)
-        try:
-            utils_misc.mount("none", tmpfs_path, "tmpfs", perm=mount_fs_size)
-            funcatexit.register(env, params.get("type"), utils_misc.umount,
-                                "none", tmpfs_path, "tmpfs")
-            process.system(numa_dd_cmd, timeout=dd_timeout, shell=True)
-        except Exception as error_msg:
-            if "No space" in str(error_msg):
-                pass
-            else:
-                test.fail("Can not allocate memory in node %s."
-                          " Error message:%s" % (numa_node_malloc,
-                                                 str(error_msg)))
-        error_context.context("Run memory heavy stress in guest", test.log.info)
-        stress_test = utils_test.VMStress(vm, "stress", params, stress_args=stress_args)
-        stress_test.load_stress_tool()
-        stress_test.unload_stress()
-        stress_test.clean()
-        utils_misc.umount("none", tmpfs_path, "tmpfs")
-        funcatexit.unregister(env, params.get("type"), utils_misc.umount,
-                              "none", tmpfs_path, "tmpfs")
-        session.cmd("sync; echo 3 > /proc/sys/vm/drop_caches")
+    node_list = host_numa_node.online_nodes
+    node_meminfo = host_numa_node.read_from_node_meminfo
+    if test_count < len(node_list):
+        test_count = len(node_list)
+    try:
+        test_mem = float(params.get("mem")) * mem_ratio
+        guest_stress_args = "-a -p -l %sM" % int(test_mem)
+        stress_path = os.path.join(data_dir.get_deps_dir('mem_mapping'), mem_map_tool)
+        test.log.info("Compile the mem_mapping tool")
+        cmd_cp_mmap_tool = cmd_cp_mmap_tool % stress_path
+        process.run(cmd_cp_mmap_tool, shell=True)
         utils_memory.drop_caches()
-
-    session.close()
+        for test_round in range(test_count):
+            cmd_mmap = params.get("cmd_mmap")
+            error_context.context("Executing stress test round: %s" % test_round, test.log.info)
+            try:
+                error_context.context("Get the qemu process memory use status", test.log.info)
+                most_used_node, memory_used = max_mem_map_node(host_numa_node, qemu_pid)
+                numa_node_malloc = most_used_node
+                mmap_size = math.floor(float(node_meminfo(numa_node_malloc, 'MemTotal')) * mem_ratio)
+                cmd_mmap = cmd_mmap % (numa_node_malloc, mmap_size)
+                error_context.context("Run mem_mapping on host node "
+                                      "%s." % numa_node_malloc, test.log.info)
+                process.system_output(cmd_mmap, shell=True, ignore_bg_processes=True)
```

Comment:
Might we call the process.system? since I didn't see you get the output to check it.
Maybe you have other concerns here.

Reply:
Nothing special, Corrected.